libs/api/repository/src/lib/gn4/platform/gn4-platform.service.ts

```diff
@@ -288,6 +296,44 @@ export class Gn4PlatformService implements PlatformServiceInterface {
     )
   }
 
+  cleanRecordAttachments(record: CatalogRecord): Observable<void> {
+    return combineLatest([
+      this.recordsApiService.getAssociatedResources(record.uniqueIdentifier),
+      this.recordsApiService.getAllResources(record.uniqueIdentifier),
+    ]).pipe(
+      map(([associatedResources, recordResources]) => {
+        const resourceIdsToKeep = [
+          ...((associatedResources as any).onlines ?? [])
+            .map((o) => o.title)
+            .map((o) => o['']),
+          ...((associatedResources as any).thumbnails ?? [])
+            .map((o) => o.title)
+            .map((o) => o['']),
+        ]
```

Comment:
What does this do? This is the kind of code that could benefit from some comments.

Reply:
Well the object returned from the API is not quite like expected so I have to cast it as any and go get the things I need. I can't put a comment above to explain yes.